src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,149 @@
+package Duke;
+
+import java.util.Scanner;
+import java.util.ArrayList;
+
+import Duke.task.Deadline;
+import Duke.task.Event;
+import Duke.task.Task;
+import Duke.exception.DukeException;
+
+public class Duke {
+	public static void main (String[] args) {
+		printStart ();
+		Scanner scan = new Scanner (System.in);
+		String input = scan.nextLine ();
+		ArrayList<Task> tasks = fileIO.readFile ();
+		int count = fileIO.getCount ();
+
+		while (!("bye".equalsIgnoreCase (input)) && !(input.isEmpty ())) {
+			try {
+				count = checkInput (input, tasks, count);
+			} catch (DukeException de) {
+				System.out.println (de.getMessage ());
+				System.out.println ("Please enter again: ");
+				printLine ();
+				continue;
+			} finally {
+				input = scan.nextLine ();
+			}
+		}
+		printEnd ();
+	}
+
+	public static int checkInput (String input, ArrayList<Task> tasks, int count) throws DukeException {
```

Comment:
This method is deeply nested. Try to extract out guard clauses so that you can reduce the indentation levels